semgrep/semgrep/cli.py

```diff
@@ -69,12 +62,20 @@ def cli() -> None:
             "name. See https://semgrep.dev/docs/writing-rules/rule-syntax for information on configuration file format."
         ),
     )
-
     config_ex.add_argument(
         "-e",
         "--pattern",
         help="Code search pattern. See https://semgrep.dev/docs/writing-rules/pattern-syntax for information on pattern features.",
     )
+    config.add_argument(
+        "-g",
+        "--generate-config",
+        action="store",
+        nargs="?",
+        const=DEFAULT_CONFIG_FILE,
+        type=argparse.FileType("x"),
```

Comment:
The `"x"` file mode only allows writing if the file doesn't already exist. This covers the existing `if Path(DEFAULT_CONFIG_FILE).exists()` check.